src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -359,6 +365,15 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		return editor.runCell(cell);
 	}
 
+	$runAllCells(id: string): Promise<boolean> {
+		let editor = this.getEditor(id);
+		if (!editor) {
+			return Promise.reject(disposed(`TextEditor(${id})`));
+		}
+		let codeCells = editor.cells.filter(cell => cell.cellType === CellTypes.Code);
```

Comment:
This is likely a duplication of code. I would prefer using the method from Model through NotebookEditor if possible. We probably need to define the best possible approach so that it will be useful going forward.